### DISCUSSION TITLE
Generating output with timesteps

### DISCUSSION THREAD

**Discussion by japresa** [idaholab/moose#18452] [Q&A General] [answered]
Hello, I'm new to MOOSE and am working on a code using SAM and MOOSE. I am running a steady state simulation and am trying to generate the output. However, I can't seem to edit the start time, end time, time steps, etc in the Executioner and Outputs blocks.

If I define parameters in the executioner block, P = {start_time, end_time, dt, num_steps}, they go unused and I'm not able to control the simulation. "/home/projects/file.i: unused parameter 'Executioner/P"

Is there a way that I can control the parameters P, in any block, to manage my output's time frame?

**Comment by GiudGiud** [comment, marked as answer]
Hello

What do you mean by this P syntax?
Also,
A steady state simulation won't take a time step parameter, it's solving for steady state not for the evolution of something over time. So if you are trying to do a relaxation transient to steady state, then you need to use a transient simulation, with a transient executioner. 

Regardless, could you please paste your executioner block here so we can have a look

Guillaume

**Reply by aeslaughter** [reply]
I am wondering if you are mixing syntax with general MOOSE and SAM. SAM likely has syntax that is specific to that application. Also, make sure you don't post input file syntax for SAM in this forum.

Perhaps @andrsd can help out here?

**Reply by andrsd** [reply]
Your executioner block should look like this:
```
[Executioner]
   start_time = <double>
   end_time = <double>
   dt = <double>
[]
```
where you need to substitute`<double>` with a floating point number. `num_steps` and `end_time` are mutually exclusive. If you specify both, `end_time` wins (IIRC).

Then, output will happen for your initial condition and every time step - that's the default. As long as you have something like this in your input file:
```
[Outputs]
  [out_displaced]
    type = Exodus
    use_displaced = true
    sequence = false
  []
[]
```

**Reply by japresa** [reply]
Thank you for your response,

The variables that @andrsd defined in the executioner block were the "parameters p" I was referring to. I will continue to model my executioner and output blocks in style you've provided.

Also, @GiudGiud, thank you for clarifying about how steady state simulations won't take time-step-parameters. I was assuming I had to analyze a transient simulation and manually gather the steady state information when reached. This software is powerful :)
